### PR TITLE
fix(#23): add daily limit check

### LINKED ISSUE #23: D2E 领取每日限额提示

### PR DESCRIPTION
**Pull request by sxlwar**
## Priority

Do you need reviewers to review this PR as soon as possible?
- [x] Medium

## Help Merge

Do you need any other people to merge it if all checks passed and reviews were approved?

- [x] Yes - Merge it immediately

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [x] No

## Why do we need this pr?

fix #23

### CODE REVIEW

File: src/components/records/D2ERecord.tsx [hunk 1]
```diff
@@ -1,82 +1,107 @@
...
-          monitor(false);
-        },
-        complete: () => {
-          observer.complete();
-          monitor(false);
-        },
-      });
+      const isRingSufficient = iif(
+        () => new BN(ring).gt(new BN(0)),
+        isSufficient(arrival as EthereumConfig, 'ring'),
```
Comment (WoeOm) [inline src/components/records/D2ERecord.tsx:48]: 需要判断（该笔待领取 RING 的数量 是否大于 当日剩余额度）
比如：今日还剩 1 RING 额度，如果该笔跨链数量为 100 RING，应该不通过 isSufficient 检验

**Review by WoeOm — CHANGES_REQUESTED** [review]

File: src/components/records/D2ERecord.tsx [hunk 2]
```diff
@@ -11,14 +11,16 @@ import { ClaimNetworkPrefix, claimToken, entrance } from '../../utils';
...
+
+function isSufficient(config: EthereumConfig, tokenType: 'ring' | 'kton', amount: BN): Observable<boolean> {
   const web3 = entrance.web3.getInstance(entrance.web3.defaultProvider);
   const store = config.contracts.e2d;
   const contract = new web3.eth.Contract(abi.tokenIssuingABI, store.redeem);
   const limit = from(contract.methods.dailyLimit(store[tokenType]).call() as Promise<string>);
   const toadySpent = from(contract.methods.spentToday(store[tokenType]).call() as Promise<string>);
 
-  return zip([limit, toadySpent]).pipe(map(([total, spent]) => new BN(total).sub(new BN(spent)).gte(new BN(0))));
+  return zip([limit, toadySpent]).pipe(map(([total, spent]) => new BN(total).sub(new BN(spent)).gte(amount)));
```
Comment (WoeOm) [inline src/components/records/D2ERecord.tsx:23]: gt(amout) ？

Comment (sxlwar) [reply src/components/records/D2ERecord.tsx:23]: 相等应该允许领吧

Comment (WoeOm) [reply src/components/records/D2ERecord.tsx:23]: 是的。我看错了，以为方法是余额不足的判断

**Review by WoeOm — APPROVED** [review]

**Pull request merged**